core/opendaq/device/include/opendaq/device.h

```diff
@@ -276,6 +277,27 @@ DECLARE_OPENDAQ_INTERFACE(IDevice, IFolder)
     virtual ErrCode INTERFACE_FUNC createDefaultAddDeviceConfig(IPropertyObject** defaultConfig) = 0;
 
     virtual ErrCode INTERFACE_FUNC getSyncComponent(ISyncComponent** sync) = 0;
+
+    /*!
+     * @brief Creates and adds to the device a server with the provided unique type ID and returns it.
+     * @param[out] server The added server.
+     * @param typeId The unique type ID of the server. Can be obtained from its corresponding Server type object.
+     * @param config A config object to configure a server with custom settings specific to that server type.
+     */
+    virtual ErrCode INTERFACE_FUNC addServer(IServer** server, IString* typeId, IPropertyObject* config) = 0;
```

Comment:
The argument order should stay the same as it was, or this breaks the API.

Reply:
I've restored the original argument order